### PR TITLE
Faet/release zetasql linux package

### PR DESCRIPTION
**Pull request by jingchen2222**
* **Please check if the PR fulfills these requirements**
    - The commit message follows our guidelines
    - Tests for the changes have been added (for bug fixes / features)
    - Docs have been added / updated (for bug fixes / features)
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature

* **What is the current behavior?** (You can also link to an open issue here)
Resolve  https://github.com/4paradigm/HybridSE/issues/60
feat: support release zetasql package in MacOS

* **What is the new behavior (if this is a feature change)?**
  - support pack zetasql in linux

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -146,53 +192,22 @@ jobs:
           source /opt/rh/rh-python38/enable
           bazel test ${{ env.build_argv }} --test_summary=detailed --test_output=errors ${{ env.target }}
 
-
   release:
-    runs-on: ubuntu-latest
```
Comment (aceforeverd) [inline .github/workflows/main.yml:151]: we should build & release on centos7 & gcc 8. `hybridsql-base` should suffice.
Other linux env exist just for build test, which can be removed if not useful

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -64,8 +64,29 @@ jobs:
       - name: Test
         run: |
           bazel test ${{ env.build_argv }} --test_summary=detailed ${{ env.target }}
+      
+      - name: Determine Version
+        # if: ${{ github.event_name == 'push' }}
```
Comment (aceforeverd) [inline .github/workflows/main.yml]: uncomment the if condition

File: pack_zetasql.sh [hunk 3]
```diff
@@ -7,75 +7,136 @@ set -eE
...
-find com_googleapis_googleapis -type f -iname '*.so' -exec bash -c 'install_external_lib $0' {} \;
+
+if [[ "$OSTYPE" == "linux-gnu"* ]]
+then
+    find icu -type f -iregex ".*/.*\.\(so\|a\)\$" -exec bash -c 'install_external_lib $0' {} \;
+    find com_googlesource_code_re2 -type f -iregex ".*/.*\.\(so\|a\)\$" -exec bash -c 'install_external_lib $0' {} \;
+    find com_googleapis_googleapis -type f -iregex ".*/.*\.\(so\|a\)\$" -exec bash -c 'install_external_lib $0' {} \;
+    find com_google_file_based_test_driver -type f -iregex ".*/.*\.\(so\|a\)\$" -exec bash -c 'install_external_lib $0' {} \;
+else
+    find icu -type f -iname '*.a' -exec bash -c 'install_external_lib $0' {} \;
```
Comment (aceforeverd) [inline pack_zetasql.sh]: the ZetaSQL archive do not include abseil-cpp on linux, instead abseil-cpp is bundled in hybridsql-asserts.
Will this ok for Mac ?

File: .github/workflows/main.yml (hunk 1 above)
Comment (jingchen2222) [reply .github/workflows/main.yml:151]: Yep. So I add `linux-build` job to build and test and pack zetasql  libraries when tag.
Other linux env exist just for build test, which can be removed if not useful

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -167,32 +136,116 @@ jobs:
...
-      - name: Find Orphan Library
+          bazel query 'deps(${{ env.target }})' |grep //zetasql|xargs bazel build ${{ env.build_argv }} 
+          bazel build "@com_googleapis_googleapis//:all" ${{ env.build_argv }}
+          bazel build "@com_google_file_based_test_driver//file_based_test_driver:all" ${{ env.build_argv }}
+          bazel build "@com_googlesource_code_re2//:re2" ${{ env.build_argv }}
+      
+      - name: Test
         run: |
-          ./find_orphan_so.sh
+          bazel test ${{ env.build_argv }} --test_summary=detailed ${{ env.target }}
```
Comment (jingchen2222) [inline .github/workflows/main.yml:150]: Here we find a way to `bazel build //zetasql/parser:all` and to build its dependent static libraries at the same time

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -167,32 +136,116 @@ jobs:
...
         run: |
           source /opt/rh/devtoolset-8/enable
           source /opt/rh/rh-python38/enable
-          bazel build ${{ env.build_argv }} ${{ env.target }}
-
-      - name: Find Orphan Library
+          bazel query 'deps(${{ env.target }})' |grep //zetasql|xargs bazel build ${{ env.build_argv }} 
+          bazel build "@com_googleapis_googleapis//:all" ${{ env.build_argv }}
+          bazel build "@com_google_file_based_test_driver//file_based_test_driver:all" ${{ env.build_argv }}
+          bazel build "@com_googlesource_code_re2//:re2" ${{ env.build_argv }}
```
Comment (jingchen2222) [inline .github/workflows/main.yml]: Here we find a way to build `//zetasql/parser:all` and to build its dependent static libraries at the same time.
`bazel query 'deps(${{ env.target }})' |grep //zetasql|xargs bazel build ${{ env.build_argv }}`

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -167,32 +135,121 @@ jobs:
...
+      - name: print toolchain info
+        run: |
+          gcc -v
+          g++ -v
+          python -V
+          java -version
+
+      - name: build
+        run: |
+          bazel build ${{ env.build_argv }} ${{ env.target }}
```
Comment (aceforeverd) [inline .github/workflows/main.yml:206]: we should create a compile script for easy reproduce on local machine

Comment (jingchen2222) [reply .github/workflows/main.yml:206]: Yep

**Pull request merged**